xbmc/utils/LangCodeExpander.cpp

```diff
@@ -1798,3 +1878,264 @@ const std::array<ISO3166_1, 245> RegionCodes = {{
     {"zw", "zwe"}
 }};
 // clang-format on
+
+// clang-format off
+// The ISO3166_1 codes above have fewer entries, are in lower-case
+// and lack the name.  Ideally, these should be merged into a
+// single source of truth and the referring functions updated
+// to account for the differences in case.
+// Source: https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2
+// Source: https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3
+// The ISO standard seems to require the codes to be upper case
+const std::array<ISO3166_1A2, 249> RegionNames = {{
+    {"AD", "AND", "Andorra"},
+    {"AE", "ARE", "United Arab Emirates"},
+    {"AF", "AFG", "Afghanistan"},
+    {"AG", "ATG", "Antigua and Barbuda"},
+    {"AI", "AIA", "Anguilla"},
```

Comment:
Seeing those English plain text names for the regions: Where will these strings be exposed? If in the user interface, all those string must be localizable.

Reply:
I don't think it's a good idea to localize them at least not now
would be a colossal job since there are also all other ISO lists that result in about 1000 strings...
in theory on GUI there should be the lookup to get ISO639 list only

Reply:
> I don't think it's a good idea to localize them at least not now
would be a colossal job since there are also all other ISO lists that result in about 1000 strings...

So, do we need these strings at all? Again, where exactly will they actually exposed to the GUI? If nowhere, drop these strings!

Reply:
if you introduce a language standard you have to add it with his complete table of values,
it is used for language code conversions and currently fwik not in GUI but for addons
this should also facilitate a future implementation of IETF BCP 47 that makes use of all these ISO standards
currently kodi is poor at recognizing languages